R/sampler-random-walk.R

```diff
@@ -7,14 +7,37 @@
 ##'
 ##' @param vcv A variance covariance matrix for the proposal.
 ##'
+##' @param boundaries Control the behaviour of proposals that are
+##'   outside the model domain.  The suported options are:
+##'
+##'   * "reflect" (the default): we reflect proposed parmeters that
+##'     lie outside the domain back into the domain (as many times as
+##'     needed)
+##'
+##'   * "ignore": evaluate the point anyway, even if it lies outside
+##'     the domain.
+##'
+##' The initial point selected will lie within the domain, as this is
+##' enforced by [mcstate_sample].
+##'
 ##' @return A `mcstate_sampler` object, which can be used with
 ##'   [mcstate_sample]
 ##'
 ##' @export
-mcstate_sampler_random_walk <- function(vcv = NULL) {
+mcstate_sampler_random_walk <- function(vcv = NULL, boundaries = "reflect") {
```

Comment:
More for thought - wondering why reflection is the default we want, rather than saying, "that one's outside of the domain, let it fail somehow"